mission_control/src/mission_control.cpp

```diff
@@ -121,20 +121,14 @@ MissionControlNode::~MissionControlNode()
 
 int MissionControlNode::loadMissionFile(std::string mission_full_path)
 {
-  MissionParser parser(node_handle);
-
   Mission* newMission = new Mission();
-  // Parse the specified mission
-  if (parser.parseMissionFile(*newMission, mission_full_path) == false)
+  if (newMission->loadBehavior(mission_full_path) == false)
   {
     delete newMission;
     return -1;
   }
```

Comment:
@patapufate if a mission needs a definition to be fully constructed, consider using factory function pattern e.g.

```c++
class Mission {
     public: 
          static Mission * FromMissionDefinition(const std::string& path);
     private:
          explicit Mission(BT::Tree);
};
```
You can better deal with errors and lifetimes.

Reply:
Done